src/openforms/formio/api/views.py

```diff
@@ -64,12 +63,22 @@ def post(self, request, *args, **kwargs):
         # trim name part if necessary but keep the extension
         name, ext = os.path.splitext(file.name)
         name = name[: 255 - len(ext)] + ext
+        mime_type = file.content_type
+        _file = file.open()  # reset the file descriptor
+        detected_mime_type = magic.from_buffer(_file.read(), mime=True)
+
+        if (
+            detected_mime_type == "application/vnd.ms-outlook"
+            and _file.content_type == "application/octet-stream"
+            and ext == ".msg"
+        ):
+            mime_type = detected_mime_type
```

Comment:
we don't need to perform the mime type validation with `magic` here any more, as that has been checked by the validation in the serializer


```suggestion
        match (mime_type := file.content_type):
            # mime type + extension validation was performed in the serializer
            case "application/octet-stream" if ext == ".msg":
                mime_type = "application/vnd.ms-outlook"
            case _:
                pass
```